segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/SegmentKeyCache.java

```diff
@@ -450,9 +458,17 @@ synchronized void update(UUID keyHash, long segmentOffset, int currentGeneration
             entryData.setLong(entryOffset, segmentOffset);
 
             // Update the cache and stats.
-            storeInCache(entryData);
-            this.generation = currentGeneration;
-            this.highestOffset = Math.max(this.highestOffset, segmentOffset);
+            try {
+                storeInCache(entryData);
+                this.generation = currentGeneration;
+                this.highestOffset = Math.max(this.highestOffset, segmentOffset);
+            } catch (CacheEntryEvictedException cex) {
+                // Pass-through exception. If this is the case, the entry has already been evicted so not more we can do.
+                throw cex;
```

Comment:
Not sure I understood why we need to throw here.
Entry doesn't exist as expected. So this should be ok. right?

Reply:
CacheEntryEvictedException is defined and thrown in this class. It's a way to handle a case when we are attempting to modify an entry while it is being evicted. Part of the process is not done while holding the cache lock so this may occur.

Reply:
ok.